modules/FindBasisUniversal.cmake

```diff
@@ -119,96 +77,163 @@ endmacro()
-                ${BasisUniversalEncoder_DIR}/basisu_gpu_texture.cpp
-                ${BasisUniversalEncoder_DIR}/basisu_pvrtc1_4.cpp
-                ${BasisUniversalEncoder_DIR}/basisu_resampler.cpp
-                ${BasisUniversalEncoder_DIR}/basisu_resample_filters.cpp
-                ${BasisUniversalEncoder_DIR}/basisu_ssim.cpp
-                ${BasisUniversalEncoder_DIR}/lodepng.cpp)
-
-            foreach(_file ${BasisUniversalEncoder_SOURCES})
-                _basis_setup_source_file(${_file})
-            endforeach()
-
-            add_library(BasisUniversal::Encoder INTERFACE IMPORTED)
-            set_property(TARGET BasisUniversal::Encoder APPEND PROPERTY
-                INTERFACE_INCLUDE_DIRECTORIES ${BasisUniversalEncoder_INCLUDE_DIR})
-            set_property(TARGET BasisUniversal::Encoder APPEND PROPERTY
-                INTERFACE_SOURCES "${BasisUniversalEncoder_SOURCES}")
-            # Explicitly *not* linking this to Threads::Threads because when
-            # done like that, std::thread creation will die on a null function
-            # pointer call (inside __gthread_create, which weakly links to
-            # pthread) in the BasisImageConverter UNLESS the final application
-            # is linked to pthread as well. As far as I tried (four hours
-            # lost), there's no way to check if the weak pthread_create pointer
-            # is null -- tried so far:
-            #  - dlsym(self, "pthread_create") returns non-null
-            #  - dlopen("libpthread.so") returns non-null
-            #  - defining a weakref pthread_create the same way as glibc does
-            #    https://github.com/gcc-mirror/gcc/blob/3e7b85061947bdc7c7465743ba90734566860821/libgcc/gthr-posix.h#L106
-            #    returns non-null
-            # The rest is documented in the BasisImageConverter plugin itself.
-            set_property(TARGET BasisUniversal::Encoder APPEND PROPERTY
-                INTERFACE_LINK_LIBRARIES BasisUniversal::Transcoder)
-            set_property(TARGET BasisUniversal::Encoder APPEND PROPERTY
-                INTERFACE_COMPILE_DEFINITIONS "BASISU_NO_ITERATOR_DEBUG_LEVEL")
         else()
             set(BasisUniversal_Encoder_FOUND TRUE)
         endif()
 
     elseif(_component STREQUAL "Transcoder")
         if(NOT TARGET BasisUniversal::Transcoder)
-            find_path(BasisUniversalTranscoder_DIR NAMES basisu_transcoder.cpp
-                HINTS "${BASIS_UNIVERSAL_DIR}/transcoder" "${BASIS_UNIVERSAL_DIR}"
-                NO_CMAKE_FIND_ROOT_PATH)
-            if(BasisUniversalTranscoder_DIR)
-                set(BasisUniversal_Transcoder_FOUND TRUE)
+            find_library(BasisUniversalTranscoder_LIBRARY_RELEASE basisu_transcoder)
+            find_library(BasisUniversalTranscoder_LIBRARY_DEBUG basisu_transcoder HINTS "debug")
+
+            if(BasisUniversalTranscoder_LIBRARY_DEBUG AND BasisUniversalTranscoder_LIBRARY_RELEASE)
```

Comment:
`OR` here, and then wrap the `set_property()` for the libraries below in corresponding `if()`s